packages/connection/src/class-error-handler.php

```diff
@@ -580,4 +589,74 @@ public function verify_xml_rpc_error( \WP_REST_Request $request ) {
 
 	}
 
+	/**
+	 * Prints a generic error notice for all connection errors
+	 *
+	 * @since 8.9.0
+	 *
+	 * @return void
+	 */
+	public function generic_admin_notice_error() {
+		// do not add admin notice to the jetpack dashboard.
+		global $pagenow;
+		if ( 'admin.php' === $pagenow || isset( $_GET['page'] ) && 'jetpack' === $_GET['page'] ) { // phpcs:ignore
+			return;
+		}
+
+		if ( ! current_user_can( 'jetpack_connect' ) ) {
+			return;
+		}
+
+		/**
+		 * Filters the message to be displayed in the admin notices area when there's a xmlrpc error
+		 *
+		 * Return an empty value to disable the message.
+		 *
+		 * @since 8.9.0
+		 *
+		 * @param string $message The error message.
+		 * @param array  $errors The array of errors. See Automattic\Jetpack\Connection\Error_Handler for details on the array structure.
+		 */
+		$message = apply_filters( 'jetpack_connection_error_notice_message', __( 'Your connection with WordPress.com seems to be broken. If you\'re experiencing issues, please try reconnecting.', 'jetpack' ), $this->get_verified_errors() );
+
+		/**
+		 * Fires inside the admin_notices hook just before displaying the error message for a broken connection.
+		 *
+		 * If you want to disable the default message from being displayed, return an emtpy value in the jetpack_connection_error_notice_message filter.
+		 *
+		 * @since 8.9.0
+		 *
+		 * @param array $errors The array of errors. See Automattic\Jetpack\Connection\Error_Handler for details on the array structure.
+		 */
+		do_action( 'jetpack_connection_error_notice', $this->get_verified_errors() );
+
+		if ( empty( $message ) ) {
+			return;
+		}
+
+		?>
+		<div class="notice notice-error is-dismissible jetpack-message jp-connect" style="display:block !important;">
+			<p><?php echo esc_html( $message ); ?></p>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Adds the error message to the Jetpack React Dashboard
+	 *
+	 * @since 8.9.0
+	 *
+	 * @param array $errors The array of errors. See Automattic\Jetpack\Connection\Error_Handler for details on the array structure.
+	 * @return array
+	 */
+	public function jetpack_react_dashboard_error( $errors ) {
+
+		$errors[] = array(
+			'code'    => 'invalid_blog_token',
+			'message' => __( 'Your connection with WordPress.com seems to be broken. If you\'re experiencing issues, please try reconnecting.', 'jetpack' ),
```

Comment:
The duplicated error message here is intentional.

Leaving it separated from the filter in the method above allows Jetpack plugin to have full control over it's Dashboard. If a plugin filters the message and modifies or disables it, it won't affect the Jetpack plugin.